docs/frameworks/nvflare.md

```diff
@@ -0,0 +1,78 @@
+# FL using NVFlare in AzureML [Experimental]
+
+**Scenario** - This tutorial is an extension of the [pneumonia / radiography classification tutorial](../real-world-examples/pneumonia.md) using the NVFlare framework instead.  
+We run an NVFlare federated learning job to detect pneumonia from chest radiographs. We mimic a real-world FL scenario where 3 hospitals in 3 different regions want to collaborate on training a model to detect pneumonia from chest radiographs. The hospitals have their own data, and they want to train a model on all data without directly sharing data with each other, or with a central entity.  
+
+:warning: Experimental :warning: This tutorial is relying on some experiment code. It will work on our quickstart sandbox, but running it outside of this controlled setup might not work. We're actively working to release a more robust and generic solution.
+
+**Dataset** - The model is trained on the [Chest X-ray dataset](https://www.kaggle.com/datasets/paultimothymooney/chest-xray-pneumonia) from Kaggle. This example is adapted from [another FL solution](https://github.com/Azure/medical-imaging/tree/main/federated-learning) by Harmke Alkemade _et al._.
+
+## Install the required dependencies
+
+You'll need python to submit experiments to AzureML. You can install the required dependencies by running:
+
+```bash
+conda env create --file ./examples/pipelines/environment.yml
+conda activate fl_experiment_conda_env
+```
+
+Alternatively, you can just install the required dependencies:
+
+```bash
+python -m pip install -r ./examples/pipelines/requirements.txt
+```
+
+## Provision an FL sandbox workspace using vnet peering
+
+To run this example, you will need to provision an AzureML workspace ready for Federated Learning.
+
+**IMPORTANT**: Provision a [quickstart eyes-off/vnet-based setup](../quickstart.md) and select `applyVnetPeering=true` to apply peering to the orchestrator and silos vnet. Without this setting, this demo will NOT work.
```

Comment:
Not directly related to the NVFlare part, but I'm getting errors when creating the setup. Whole deployment is [here](https://ms.portal.azure.com/#view/HubsExtension/DeploymentDetailsBlade/~/overview/id/%2Fsubscriptions%2F48bbc269-ce89-4f6f-9a12-c6f91fcb772d%2FresourceGroups%2Fthopo-owner-fldev-rg%2Fproviders%2FMicrosoft.Resources%2Fdeployments%2FMicrosoft.Template-20230123160232/packageId/Microsoft.Template/packageIconUri/https%3A%2F%2Fcatalogartifact.azureedge.net%2Fpublicartifacts%2FMicrosoft.Template-2.1.8%2FMedium.png/assetTypeId/Microsoft_Azure_CreateUIDef%3Eundefined/primaryResourceId//createBlade~/%7B%22bladeName%22%3A%22CustomDeploymentBlade%22%2C%22extension%22%3A%22Microsoft_Azure_CreateUIDef%22%2C%22telemetryId%22%3A%22049c67b4-117c-4138-b26f-84a78150ad09%22%7D), and errors can be seen [here](https://ms.portal.azure.com/#view/HubsExtension/DeploymentDetailsBlade/~/overview/id/%2Fsubscriptions%2F48bbc269-ce89-4f6f-9a12-c6f91fcb772d%2FresourceGroups%2Fthopo-owner-fldev-rg%2Fproviders%2FMicrosoft.Resources%2Fdeployments%2Fnvflare-test-vnetpeering-orch-to-silo1) or [there](https://ms.portal.azure.com/#view/HubsExtension/DeploymentDetailsBlade/~/overview/id/%2Fsubscriptions%2F48bbc269-ce89-4f6f-9a12-c6f91fcb772d%2FresourceGroups%2Fthopo-owner-fldev-rg%2Fproviders%2FMicrosoft.Resources%2Fdeployments%2Fnvflare-test-vnetpeering-orch-to-silo0). Any recommendation @jfomhover ? It seems like my orchestrator VNet does not have a Gateway and I'm not sure how to fix

Reply:
I didn't want to wait longer so now I'm using [this already-provisioned workspace](https://ml.azure.com/?wsid=/subscriptions/48bbc269-ce89-4f6f-9a12-c6f91fcb772d/resourceGroups/vertical-fl-sandbox-rg/providers/Microsoft.MachineLearningServices/workspaces/aml-vfl&tid=72f988bf-86f1-41af-91ab-2d7cd011db47), which already has VNet peering. Then I'm not able to submit the jobs, getting an error about environment. Let's talk offline @jfomhover

Reply:
Was able to submit with environment from the client component (it was missing from the provision component)

Reply:
Please try from the release-05 branch, the provisioning script from main had an issue that I fixed in a PR in the release-05 branch.

Reply:
Provisioning all good now, my bad for using the template in main instead of release branch. There is still the issue about the environment for provision component though

Reply:
Tests by Amit confirmed the job is working seamlessly, I'll mark this one as resolved.